src/c/profile/discovery/transport/udp_transport_datagram_posix_nopoll.c

```diff
@@ -0,0 +1,94 @@
+#include "udp_transport_datagram_internal.h"
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+
+bool uxr_init_udp_transport_datagram(
+        uxrUDPPlatform* platform,
+        uxrIpProtocol ip_protocol,
+        const char* ip,
+        const char* port)
+{
+    int fd = socket(PF_INET, SOCK_DGRAM, 0);
+    transport->fd = fd;
+
+    return fd != -1;
+}
+
+bool uxr_close_udp_transport_datagram(
+        uxrUDPTransportDatagram* transport)
+{
+    return (0 == close(transport->poll_fd.fd));
+}
+
+bool uxr_udp_send_datagram_to(
+        uxrUDPTransportDatagram* transport,
+        const uint8_t* buf,
+        size_t len,
+        const TransportLocator* locator)
+{
+    bool rv = true;
+    switch (locator->format)
+    {
+        case ADDRESS_FORMAT_MEDIUM:
+        {
+            struct sockaddr_in remote_addr;
+            memcpy(&remote_addr.sin_addr, locator->_.medium_locator.address, sizeof(remote_addr.sin_addr));
+            remote_addr.sin_family = AF_INET;
+            remote_addr.sin_port = htons(locator->_.medium_locator.locator_port);
+
+            ssize_t bytes_sent = sendto(transport->poll_fd.fd, (const void*)buf, len, 0,
+                                        (struct sockaddr*)&remote_addr, sizeof(remote_addr));
+            if (0 > bytes_sent)
+            {
+                rv = false;
+            }
+            break;
+        }
+        default:
+            rv = false;
+            break;
+    }
+    return rv;
+}
+
+bool uxr_udp_recv_datagram(
+        uxrUDPTransportDatagram* transport,
+        uint8_t** buf,
+        size_t* len,
+        int timeout)
+{
+    bool rv = false;
+
+    timeout = (timeout <= 0) ? 1 : timeout;
+
+    struct timeval tv;
+    tv.tv_sec = timeout / 1000;
+	tv.tv_usec = (timeout % 1000) * 1000;
+
+    setsockopt(platform->fd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv));
```

Comment:
```suggestion
    setsockopt(transport->fd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv));
```

Reply:
Since tests passed, I assume we are not covering all possible test cases here. @pablogs9 do you know if transports are being tested elsewhere? (such as in Micro XRCE-DDS general repo). Here I can see tests only for serial transport. If that is not the case, we sould come up with a more complete test battery that allows detecting these issues.